src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -273,18 +273,111 @@ public void Shuffle<T>(T[] values)
         /// </remarks>
         public void Shuffle<T>(Span<T> values)
         {
-            int n = values.Length;
+            var chooser = new IncreasingUniform(this);
+            for (int i = 1; i < values.Length; i++)
+            {
+                int index = chooser.NextIndex();
+                (values[i], values[index]) = (values[index], values[i]);
```

Comment:
Does https://github.com/dotnet/runtime/pull/70944#discussion_r962206857 apply here?
In the linked sharplabs there's still a (little) difference in generated IL.